app/blueprints/assessments/activity_trail.py

```diff
@@ -0,0 +1,307 @@
+                    current_app.logger.info(
+                        "Adding flagged sections to RAISED flag"
+                    )
+                    result.append(updated_update)
+                else:
+                    result.append(update)
+        return result
+
+    @classmethod
+    def from_list(cls, flags_list: list[dict]):
+        if flags_list:
+            flags = cls.process_flags(flags_list)
+            return super().from_list(flags)
+        else:
+            return []
+
+
+@dataclass
+class Comments(BaseModel):
+    id: str
+    comment: str
+    comment_type: str
+    date_created: str
+    sub_criteria_id: str
+    theme_id: str
+    user_id: str
+    full_name: str = ""
+    highest_role: str = ""
+    email_address: str = ""
+
+    @classmethod
+    def process_comments(cls, comments_list: list) -> list[dict]:
+        """
+        Retrieve all comments from updates and add the following attributes to each comment:
+        - "user_id"
+        - "sub_criteria_id"
+        - "theme_id"
+        - "comment_type"
+        - "full_name"
+        - "email_address"
+        - "highest_role"
+
+        Args: comments_list (list): A list of comments.
+
+        Returns:
+            list[dict]: A list of dictionaries containing comments with additional attributes.
+        """
+        result = []
+
+        for comment in comments_list:
+            for update in comment.get("updates"):
+                updated_update = {
+                    **update,
+                    "user_id": comment.get("user_id"),
+                    "sub_criteria_id": comment.get("sub_criteria_id"),
+                    "theme_id": comment.get("theme_id"),
+                    "comment_type": comment.get("comment_type"),
+                    "full_name": comment.get("full_name"),
+                    "email_address": comment.get("email_address"),
+                    "highest_role": comment.get("highest_role"),
+                }
+                current_app.logger.info(
+                    "Adding required information to comment"
+                )
+                result.append(updated_update)
+
+        return result
+
+    @classmethod
+    def from_list(cls, comments_list: list[dict]):
+        if comments_list:
+            comments = cls.process_comments(comments_list)
+            return super().from_list(comments)
+        else:
+            return []
+
+
+@dataclass
+class Scores(BaseModel):
+    application_id: str
+    date_created: str
+    id: str
+    justification: str
+    score: int
+    sub_criteria_id: str
+    user_id: str
+    full_name: str = ""
+    highest_role: str = ""
+    email_address: str = ""
+
+
+def get_user_info(list_data: list, state: AssessorTaskList) -> dict:
+
+    if list_data is None:
+        return []
+
+    user_list = []
+    for item in list_data:
+        user_id = item.user_id
+        if user_id is not None and user_id not in user_list:
+            user_list.append(user_id)
```

Comment:
You could use a set comprehension for this, since it doesn't contain duplicates

```suggestion
    user_ids = {item.user_id for item in list_data if item.user_id}
    # user_list = list(user_ids)  # if you need it as a list afterwards
```

Reply:
Aah yes! Thank you!